app/src/main/java/org/thoughtcrime/securesms/onboarding/recoverypassword/RecoveryPasswordActivity.kt

```diff
@@ -0,0 +1,218 @@
+            modifier = Modifier
+                .contentDescription(R.string.AccessibilityId_recovery_password)
+                .verticalScroll(rememberScrollState())
+                .padding(bottom = 16.dp)
+        ) {
+            SmallButtons {
+                RecoveryPasswordCell(seed, copySeed)
+                HideRecoveryPasswordCell(onHide)
+            }
+        }
+    }
+}
+
+@Composable
+fun RecoveryPasswordCell(seed: String, copySeed:() -> Unit = {}) {
+    var showQr by remember {
+        mutableStateOf(false)
+    }
+
+    CellWithPaddingAndMargin {
+        Column {
+            Row {
+                Text(stringResource(R.string.sessionRecoveryPassword))
+                Spacer(Modifier.width(8.dp))
+                SessionShieldIcon()
+            }
+
+            Text(stringResource(R.string.recoveryPasswordDescription))
+
+            AnimatedVisibility(!showQr) {
+                Text(
+                    seed,
+                    modifier = Modifier
+                        .contentDescription(R.string.AccessibilityId_recovery_password_container)
+                        .padding(vertical = 24.dp)
+                        .border(
+                            width = 1.dp,
+                            color = classicDarkColors[3],
+                            shape = RoundedCornerShape(11.dp)
+                        )
+                        .padding(24.dp),
+                    textAlign = TextAlign.Center,
+                    style = MaterialTheme.typography.small.copy(fontFamily = FontFamily.Monospace),
+                    color = MaterialTheme.colors.run { if (isLight) onSurface else secondary },
+                )
+            }
+
+            AnimatedVisibility(
+                showQr,
+                modifier = Modifier.align(Alignment.CenterHorizontally)
+            ) {
+                QrImage(
+                    seed,
+                    modifier = Modifier.padding(vertical = 24.dp)
+                        .contentDescription(R.string.AccessibilityId_qr_code),
+                    icon = R.drawable.session_shield
+                )
+            }
+
+            AnimatedVisibility(!showQr) {
+                Row(horizontalArrangement = Arrangement.spacedBy(32.dp)) {
+                    OutlineTemporaryStateButton(
+                        Modifier
+                            .weight(1f)
+                            .contentDescription(R.string.AccessibilityId_copy_button),
+                        onClick = copySeed
+                    ) { isTemporary ->
+                        Text(stringResource( if (isTemporary) R.string.copied else R.string.copy))
+                    }
+                    OutlineButton(textId = R.string.qrView, modifier = Modifier.weight(1f), onClick = { showQr = !showQr })
+                }
+            }
+
+            AnimatedVisibility(showQr, modifier = Modifier.align(Alignment.CenterHorizontally)) {
+                OutlineButton(
```

Comment:
The text in that component is not style so it does not match the designs.
Furthermore, these specific two outlined buttons are smaller, with a different text style than the other outlined buttons.
The normal OutlineButton should be modified to have explicit text styling (to base), and these buttons should probably be another Composable (OutlineButtonSmall?), with their text set to extraSmall bold